### ISSUE TITLE
Resolving Gradle Plugins from Maven Repositories using the Plugins DSL

### ISSUE THREAD

**Issue by aspin** [gradle/gradle#1541]
### Expected Behavior

Step 1: Added Maven Central to settings.gradle for pluginRepositories.
```groovy
// settings.gradle
pluginRepositories {
    gradlePluginPortal()
    maven {
        url 'https://repo1.maven.org/maven2/'
    }
}
```

Step 2: Add a plugin that's only found in Maven Central and not the Gradle Central Plugin Repository.
```groovy
// build.gradle
plugins {
    id 'se.transmode.gradle.gradle-docker' version '1.2'
}
```
Step 3: Use the plugin!

### Current Behavior
Actual Step 3: Be very confused about how gradle searches for `[plugin-id]:[plugin-id].gradle.plugin:[version]` in an attempt to resolve the artifact. Try to put in the raw maven dependency (`se.transmode.gradle:gradle-docker:1.2`), but get told that colons aren't allowed. I'm not sure if this was a convention that gradle is trying to enforce on gradle plugins created in Maven, but at the time of writing basically no plugins in Maven Central meet this naming convention.

### Context
I understand the strict naming as a new convention for the Plugins DSL but it just totally breaks the ability to import gradle plugins from Maven repos we don't control (unless I'm just doing this really wrong). Obviously I could switch to buildscript / apply plugin but the rest of the plugins in this project are already using the Plugins DSL :(

### Your Environment
Mac OS X, IntelliJ Idea

**Comment by bmuschko** [comment]
A plugin can only be resolved from a [custom repository](https://docs.gradle.org/current/userguide/plugins.html#sec:custom_plugin_repositories) (one that is not the Gradle plugin portal) if it was published with the Java plugin development plugin. The Java plugin development plugin can create a [plugin marker artifacts](https://docs.gradle.org/current/userguide/plugins.html#sec:plugin_markers) which translates to the ID you use in the plugin DSL.

The plugin you are trying to resolve does not provide use a plugin marker artifact. At the moment the underlying Maven publish plugin does not support signing of artifacts which is required for publishing to Maven Central.

The bottom line is that you have to fall back to `buildscript` in this case.
